mkosi/__init__.py

```diff
@@ -148,13 +148,24 @@ class MkosiException(Exception):
     """Leads to sys.exit"""
 
 
+class ColorPrinter():
+    isatty = sys.stderr.isatty()
+    prefix = '‣ '
+    bold = '\033[0;1;39m' if isatty else ''
+    reset = '\033[0m' if isatty else ''
```

Comment:
This is an an excellent change, but I think it could go a bit further. I think it would make sense to subsume all calls to `sys.stderr.write` (mostly in the summary) in a `MkosiPrinter` class and maybe also get rid of the calls to `print()` here and there, so that everything goes through the same machinery.

Reply:
I tend to agree here, if we do this, let's do it properly.

Reply:
any thoughts on going through the python logging facilities? That'd be an easy option to control formatting and have a distinct stream of outputs.

Reply:
Definitely open to it. I think I still have a branch lying around where I had a go at it, but I was unhappy with the log levels and messing around with handlers to remove the terminal control sequences when not needed. I think moving everything to a single printer would be a sensible step to move to logging later.

Reply:
fixed in the latest revision with the `MkosiPrinter` class, we can swap that out for a python logger in the future.